### ISSUE TITLE
"prefer-const" should not complain when I (have to) declare a variable in an outer scope but the only assignment is inner scope

### ISSUE THREAD

**Issue by WhyEvenTry** [eslint/eslint#5283]
**EDIT:** See comments below and #5251 - ticket changed to 

"Please amend the error message to include that the variable is not used anywhere else and can be moved inside the function"

Or as @mysticatea says under #5251 :

> Hm, this behavior may be good to be split out to something like new prefer-small-scope rule. I noticed this rule is doing two things, "detecting it's never modified" and "detecting it can be moved".

<br>

-----

<br>

**What version are you using?**
eslint 2.0.0 (for node.js 5.6 code using ES 2015)

**What did you do?**
Create a test file with the code below. The point is that the only assignment takes place inside the function - but for visibility reasons I *must* declare the variable in the outer scope. Which means I can't use  `const` - I cannot assign a value at the point of declaration.

```
'use strict';

let outerScopeVar;

/**
 * @returns {undefined}
 */
function myFn () {
    outerScopeVar = 'foo';
    return outerScopeVar;
}

myFn();

// Other functions using the value assigned to the variable,
// in my real code it's various async (promise) functions using the variable
//....
```

**What happened?**

$ eslint ../Test/lib/test.js

```
C:\Users\Michael\OneDrive\Projects\Test\lib\test.js
  3:5  error  'outerScopeVar' is never modified, use 'const' instead  prefer-const

✖ 1 problem (1 error, 0 warnings)
```

**Comment by BYK** [comment]
Looks like a potential bug to me. @eslint/eslint-team ?

**Comment by mysticatea** [comment]
Thank you for this issue.

I guess this is the same as #5251.

**Comment by WhyEvenTry** [comment]
Yes indeed.

**Comment by WhyEvenTry** [comment]
I reopened because #5251 was closed, but it's valid.

**Comment by WhyEvenTry** [comment]
Actually - everything already said in #5251 is correct, but in this case I would like to change the ticket to instead say something like

"Please amend the error message to include that the variable is not used anywhere else and can be moved inside the function"

Otherwise I expect more people will look at it as an error.

**Comment by BYK** [comment]
I think the rule should simply check for any nested contexts or not warn here. Also I support @mysticatea's suggestion in https://github.com/eslint/eslint/issues/5251#issuecomment-183829644 for a "prefer smaller scope" rule.

**Comment by mysticatea** [comment]
I opened the issue: #5284

**Comment by WhyEvenTry** [comment]
I don't think my issue is needed any more, so I'll close it (again).

**Comment by mysticatea** [comment]
@WhyEvenTry I'm sorry to take your time.

**Comment by WhyEvenTry** [comment]
@mysticatea I don't understand, misunderstanding? I just think my ticket is obsolete now that there is one describing the actual issue much better.
